src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1132,7 +1132,7 @@ void QgsPostgresProvider::setEditorWidgets()
   }
 
   QStringList quotedFnames;
-  foreach ( const QString &name, mAttributeFields.names() )
+  for ( const QString &name : mAttributeFields.names() )
```

Comment:
This will detach

Reply:
Where can I read more about this "detach" thing ?
I'm iterating it as const (you mentioned a condition being iterating as non-const?).
Also, why   const QStringList fieldNames = mAttributeFields.names() instead of holding a const reference  ? Let's take the chance to see if test will fail with current code ?

Reply:
btw, I'm sure there's a more elegant way to call a function for each element in a container...

Reply:
Tests won't fail. It's just a matter of loosing some memory and cpu cycles.
There's a thorough explanation here: https://www.kdab.com/goodbye-q_foreach/

Reply:
This one looks nice https://stackoverflow.com/questions/10516495/calling-a-function-on-every-element-of-a-c-vector I've never tested it though